src/libcore/pin.rs

```diff
@@ -11,13 +11,13 @@
 //! until it gets dropped. We say that the pointee is "pinned".
 //!
 //! By default, all types in Rust are movable. Rust allows passing all types by-value,
-//! and common smart-pointer types such as `Box<T>` and `&mut T` allow replacing and
-//! moving the values they contain: you can move out of a `Box<T>`, or you can use [`mem::swap`].
-//! [`Pin<P>`] wraps a pointer type `P`, so `Pin<Box<T>>` functions much like a regular `Box<T>`:
-//! when a `Pin<Box<T>>` gets dropped, so do its contents, and the memory gets deallocated.
-//! Similarly, `Pin<&mut T>` is a lot like `&mut T`. However, [`Pin<P>`] does not let clients
-//! actually obtain a `Box<T>` or `&mut T` to pinned data, which implies that you cannot use
-//! operations such as [`mem::swap`]:
+//! and common smart-pointer types such as [`Box<T>`] and `&mut T` allow replacing and
+//! moving the values they contain: you can move out of a [`Box<T>`], or you can use
+//! [`mem::swap`]. [`Pin<P>`] wraps a pointer type `P`, so [`Pin`]`<`[`Box<T>`]`>` functions much
```

Comment:
It would be nice if intra docs understood the concept of a type constructor and gave you two links automatically.

Reply:
Type constructor? You're talking about `Pin<Box<T>>` I guess? If so, we can't be sure you want two links or just one. :)

Reply:
Two links, one to `Pin` and one to `Box`. We don't want a link to `T` because it's a type variable (well... it might be useful actually...).

Reply:
It \*could\*, we can't guess it automatically unfortunately.

Reply:
@GuillaumeGomez Can you not use resolution info check whether `T` comes from a type variable? For `Pin<Box<T>>` we can at least determine syntactically that `Pin` and `Box` are type constructors and that `Pin`, `Box`, and `T` are separate entities.

Reply:
Yes but we can't be sure we're referring to an actual type or not something we just made up in the explanation two lines ago. :-/